EIPS/eip-5920.md

```diff
@@ -0,0 +1,51 @@
+---
+eip: 5920
+title: PAY opcode
+description: Introduces a new opcode, PAY, to send ether to an address without calling it
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/eip-5920-pay-opcode/11717
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-03-14
+---
+
+## Abstract
+
+This EIP adds a new opcode, `PAY(addr, val)`, that transfers `val` wei to the address `addr` without calling it.
+
+## Motivation
+
+Currently, to send ether to an address requires you to call a function of that address, which has a few issues. First of all, it opens a reentrancy attack vector, as the recipient can call back into the sender. Secondly, it opens a DoS vector, so parent functions must be cognizant of the possibility that the recipient will run out of gas or revert. Finally, the `CALL` opcode is needlessly expensive for simple ether transfers, as it requires the memory and stack to be expanded, the recipient's full data including code and memory to be loaded, and finally needs to execute a call, which might do other unintentional operations. Having a dedicated opcode for ether transfers solves all of these issues, and would be a useful addition to the EVM.
+
```

Comment:
[peer review] Adding a motivation.

```suggestion
Currently, to send ether to an address requires you to call a function of that address, which has a few issues. First of all, it opens a reentrancy attack vector, as the recipient can call back into the sender. Secondly, it opens a DoS vector, so parent functions must be cognizant of the possibility that the recipient will run out of gas or revert. Finally, the `CALL` opcode is needlessly expensive for simple ether transfers, as it requires the memory and stack to be expanded, the recipient's full data including code and memory to be loaded, and finally needs to execute a call, which might do other unintentional operations. Having a dedicated opcode for ether transfers solves all of these issues, and would be a useful addition to the EVM.

Today, to achieve such behavior without core change, solidity introduced lang feature such as nonpayable unless marked as`payable`. This shows the necessity for having such safer measure, but solidity's protection measures unfortunately adds extra gas cost overhead. Having a core-level support reduces gas by potentially significant amount <!-- TODO @Pandapip1 consider add analysis on how much gas it safes -->, which increase global ETH gas efficiency.
```

Reply:
Why does marking an address as payable increase the gas cost? I thought it was just a compiler protection.

Reply:
It's not a compiler protection, there is actually line of code added to do check
Let me find the link.

My point here is that, the fact Solidity does so, demonstrates a strong need

Reply:
https://github.com/ethereum/solidity/blob/5f8b48793cdf9a41d6f16fa601250987bee82e3e/libsolidity/codegen/ContractCompiler.cpp#L429

Citing my own answer in 
https://github.com/ethereum/EIPs/pull/5805#discussion_r1006934757

Reply:
I'm pretty sure that line of code refers to adding the check to make sure that non-payable functions aren't called with any `msg.value`, which is used specifically so that wallets can automatically call the correct payable function. That additional bytecode would still be usefully included.

Reply:
Yeah, it's still useful to be included. For what I mean is that it shows a `Motivation` to decouple PAY vs CALL with value sometimes.

Reply:
I'm still not convinced.